pkg/server/plugin/keymanager/gcpkms/client_fake.go

```diff
@@ -324,11 +330,35 @@ func (k *fakeKMSClient) setIsKeyDisabled(ok bool) {
 	k.keyIsDisabled = ok
 }
 
-func (k *fakeKMSClient) setGetPublicKeyErr(fakeError error) {
+func (k *fakeKMSClient) setGetPublicKeySequentialErrs(fakeErrors ...error) {
 	k.mu.Lock()
 	defer k.mu.Unlock()
+	errsRing := ring.New(len(fakeErrors))
```

Comment:
Are you using the ring buffer so you don't have to track the index?

Reply:
Exactly, i wanted a circular linked list behavior for the errors we can configure to be returned so that the errors will be returned sequentially, and we don't actually need to configure the returned error for every call.

Reply:
Since you're not actually using the ring functionality and wrapping around, i think a better approach is the one used  in fakedatastore.go that gives you the same benefit: https://github.com/spiffe/spire/blob/84866e1b677c8f8da997fd49d2f7ae43f0a3119e/test/fakes/fakedatastore/fakedatastore.go#L322-L329

Reply:
yeah, this approach will give me the sequential order for the errors, but not the circular behavior. I will have to update other tests (maybe just one another that reaches the max attempts for retrieving the public key ) but I think it will be worth having a simple code here 😄

Reply:
632d64e